cmd/common/utils/flags.go

```diff
@@ -187,7 +188,8 @@ func ParseEarlyFlags(cmd *cobra.Command) error {
 		cmd.FParseErrWhitelist.UnknownFlags = false
 	}()
 	// temporary workaround for https://github.com/inspektor-gadget/inspektor-gadget/pull/2174#issuecomment-1780923952
-	args := removeSplitSortArgs(os.Args[1:])
+	args := slices.Clone(os.Args) // clone to avoid modifying the original os.Args
+	args = removeSplitSortArgs(args)
```

Comment:
Good catch!

This is another solution but I think your proposal is cleaner:

```go
// removeSplitSortArgs removes the --sort flag with its arg, if it isn't in the
// merged form of --sort=foo
func removeSplitSortArgs(args []string) []string {
	for i := 0; i < len(args); i++ {
		if args[i] == "--sort" {
			// Remove also the next element as it is the arg of --sort
			ret := make([]string, 0, len(args)-2)
			ret = append(ret, args[:i]...)
			return append(ret, args[i+2:]...)
		}
	}
	return args
}

// removeHelpArg removes the --help flag
func removeHelpArg(args []string) []string {
	for i := 0; i < len(args); i++ {
		if args[i] == "--help" || args[i] == "-h" {
			ret := make([]string, 0, len(args)-1)
			ret = append(ret, args[:i]...)
			return append(ret, args[i+1:]...)
		}
	}
	return args
}
```

Reply:
> This is another solution but I think your proposal is cleaner

Yeah, Also cloning one level above ensure it is done for all following function.